File: Java/src/main/java/QuestionBank/Solution001_100/Solution67.java
package QuestionBank.Solution001_100;

/**
 * @author windows
 */
public class Solution67 {

    public String addBinary(String a, String b) {
        int indexA = a.length() - 1;
        int indexB = b.length() - 1;
        int carry = 0;
        StringBuilder sb = new StringBuilder();
        while(indexA >= 0 || indexB >= 0){
            int aa = indexA >= 0 ? a.charAt(indexA) - '0' : 0;
            int bb = indexB >= 0 ? b.charAt(indexB) - '0' : 0;
            int sum = aa + bb + carry;
            sb.append(sum % 2);
            carry = sum / 2;
            indexA--;
            indexB--;
        }
        if (carry != 0){
            sb.append(carry);
        }
        return sb.reverse().toString();
    }

    public static void main(String[] args) {
        Solution67 solution67 = new Solution67();
        System.out.println(solution67.addBinary("1010", "1011"));
    }
}

File: C++/src/Solution1_100/Solution67.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution67.h"
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;

string Solution67::addBinary(string a, string b) {
    int carry = 0;
    string result = "";
    int i = a.size() - 1, j = b.size() - 1;
    while (i >= 0 || j >= 0 || carry > 0){
        int sum = carry;
        if (i >= 0){
            sum += a[i--] - '0';
        }
        if (j >= 0){
            sum += b[j--] - '0';
        }
        carry = sum / 2;
        result = char (sum % 2 + '0') + result;
    }
    return result;
}
